File: Billy_Clara_Code/Assignments/JavaHW/bubblesort/BubbleSort.java
package com.billy.bubblesort;

public class BubbleSort {
	
	void bubbleSort(int arr[]){
		
		int n = arr.length;
		for(int i = 0; i <= n; i++) {
			
			for(int j = 0; j < n-1; j++)
			if (arr[j] > arr[j+1]) {
				
			int temp = arr[j];
			arr[j] = arr[j+1];
			arr[j+1] = temp;
			
			}
		}
	}
	
	void printArr(int arr[]) {
     
        int n = arr.length; 
        for (int i=0; i<n; ++i) 
            System.out.print(arr[i] + " "); 
        System.out.println(); 
    }
	
	public static void main(String[] args) { 
        BubbleSort BS = new BubbleSort(); 
        int arr[] = {1, 0, 5, 6, 3, 2, 3, 7, 9, 8, 4}; 
        BS.bubbleSort(arr); 
        BS.printArr(arr); 
    } 
}

File: Angel_Cruz_Code/MYASSIGNMENTS/Javascript/BubbleSort.js
// 2. Bubble Sort
// Define function: bubbleSort(numArray)
// Use the bubble sort algorithm to sort the array.
// Return the sorted array.

function bubbleSort(array) {
    var newArr = array.split(', ');
    var newArr2 = newArr.map(Number);

    var swapped;
    do {
        swapped = false;
        for (var i=0; i < newArr2.length-1; i++) {
            if (newArr2[i] > newArr2[i+1]) {
                var temp = newArr2[i];
                newArr2[i] = newArr2[i+1];
                newArr2[i+1] = temp;
                swapped = true;
            }
        }
    }
    while (swapped);
    document.getElementById("output").innerHTML=newArr2;
    return;
}